Sources/ApolloCodegenLib/Frontend/CompilationResult.swift

```diff
@@ -0,0 +1,184 @@
+import JavaScriptCore
+
+/// The output of the frontend compiler.
+public struct CompilationResult: Decodable {
```

Comment:
We only need the ability to decode. And because there's quite a bit of custom code for decoding enums and schema types from type names, I didn't want to unnecessarily have to write that code for encoding as well. (We couldn't just a make a few of the structs `Encodable` because that means all its properties would also have to be `Encodable`.)

Reply:
Interesting. I was thinking maybe there was some way to serialize the already-parsed bits so we wouldn't have to rerun the JS every time, but I suppose since we're handing stuff to the JS runtime it still needs to do that, right?

Reply:
Yeah, I don't think that would buy us much. Parsing is pretty fast. And you'd still need access to the schema. What we do in the VS Code extension is keep parsed documents in memory and only reparse the documents that change. If Xcode ever extends its plugin capabilities perhaps we could do something similar to get parsing/validation as-you-type.